From: Ray Strode <rstrode@redhat.com>
Date: Tue, 6 Feb 2024 12:59:56 -0500
Subject: loginDialog: Move authPrompt.begin to helper function

Soon, we're going to need to do some other things than firing
off the auth prompt when beginning verification.

In prepration for that, this commit creates a wrapper function,
_beginVerification, that can be easily augmented later.
---
 js/gdm/loginDialog.js | 19 +++++++++++++++----
 1 file changed, 15 insertions(+), 4 deletions(-)

diff --git a/js/gdm/loginDialog.js b/js/gdm/loginDialog.js
index 09364d6..f63a6ff 100644
--- a/js/gdm/loginDialog.js
+++ b/js/gdm/loginDialog.js
@@ -38,6 +38,7 @@ import * as LoginManager from '../misc/loginManager.js';
 import * as Main from '../ui/main.js';
 import * as MessageTray from '../ui/messageTray.js';
 import * as ModalDialog from '../ui/modalDialog.js';
+import * as Params from '../misc/params.js';
 import * as PopupMenu from '../ui/popupMenu.js';
 import * as Realmd from './realmd.js';
 import * as UserWidget from '../ui/userWidget.js';
@@ -1116,7 +1117,7 @@ export const LoginDialog = GObject.registerClass({
         if (previousUser && beginRequest === AuthPrompt.BeginRequestType.REUSE_USERNAME) {
             this._user = previousUser;
             this._authPrompt.setUser(this._user);
-            this._authPrompt.begin({userName: previousUser.get_user_name()});
+            this._beginVerification({userName: previousUser.get_user_name()});
         } else if (beginRequest === AuthPrompt.BeginRequestType.PROVIDE_USERNAME) {
             if (!this._disableUserList)
                 this._showUserList();
@@ -1188,7 +1189,7 @@ export const LoginDialog = GObject.registerClass({
                 const answer = this._authPrompt.getAnswer();
                 this._user = this._userManager.get_user(answer);
                 this._authPrompt.clear();
-                this._authPrompt.begin({userName: answer});
+                this._beginVerification({userName: answer});
                 this._updateCancelButton();
             });
         this._updateCancelButton();
@@ -1480,6 +1481,15 @@ export const LoginDialog = GObject.registerClass({
         });
     }
 
+    _beginVerification(params) {
+        params = Params.parse(params, {
+            userName: null,
+            hold: null,
+        });
+
+        this._authPrompt.begin(params);
+    }
+
     _setUserListExpanded(expanded) {
         this._userList.updateStyle(expanded);
         this._userSelectionBox.visible = expanded;
@@ -1498,7 +1508,7 @@ export const LoginDialog = GObject.registerClass({
 
     _hideUserListAndBeginVerification() {
         this._hideUserList();
-        this._authPrompt.begin();
+        this._beginVerification();
     }
 
     _showUserList() {
@@ -1518,7 +1528,8 @@ export const LoginDialog = GObject.registerClass({
         const userName = item.user.get_user_name();
         const hold = new Batch.Hold();
 
-        this._authPrompt.begin({userName, hold});
+        this._beginVerification({userName, hold});
+
         return hold;
     }
 
